northstar/src/runtime/fork/forker/impl.rs

```diff
@@ -98,10 +76,16 @@ pub async fn run(stream: StdUnixStream, notifications: StdUnixStream) -> ! {
                     Some(Message::ExecRequest { container, path, args, env, io }) => {
                         let io = io.unwrap();
                         if let Some(init) = inits.remove(&container) {
+                            // There's a init - let's exec!
                             let (response, exit) = exec(init, container, path, args, env, io).await;
-                            tx.send(exit).await.ok();
+
+                            // Add exit status future of this exec request
+                            exits.push(exit);
+
+                            // Send the result of the exec request to the runtime
                             stream.send(response).await.expect("failed to send response");
                         } else {
+                            // TODO: This is a bug, we should never get here
```

Comment:
Then we should probably call `expect` after the `.remove` and not handle this impossible case

Reply:
Agree. That situation cannot be handled in any good way. I updated the two branches to panic if something is wrong.